network/src/noise/handshake.rs

```diff
@@ -481,41 +497,61 @@ mod test {
 
     /// helper to perform a noise handshake with two peers
     fn perform_handshake(
-        client: NoiseUpgrader,
-        server: NoiseUpgrader,
+        client: &NoiseUpgrader,
+        server: &NoiseUpgrader,
         server_public_key: x25519::PublicKey,
     ) -> (
-        NoiseStream<MemorySocket>,
-        (NoiseStream<MemorySocket>, PeerId),
+        io::Result<NoiseStream<MemorySocket>>,
+        io::Result<(NoiseStream<MemorySocket>, PeerId)>,
     ) {
         // create an in-memory socket for testing
         let (dialer_socket, listener_socket) = MemorySocket::new_pair();
 
         // perform the handshake
-        let (client_session, server_session) = block_on(join(
+        block_on(join(
             client.upgrade_outbound(dialer_socket, server_public_key),
             server.upgrade_inbound(listener_socket),
-        ));
-
-        (client_session.unwrap(), server_session.unwrap())
+        ))
     }
 
     fn test_handshake_success(is_mutual_auth: bool) {
         // perform handshake with two testing peers
         let ((client, client_public), (server, server_public)) = build_peers(is_mutual_auth);
-        let (client, (server, _)) = perform_handshake(client, server, server_public);
 
-        assert_eq!(client.get_remote_static(), server_public);
-        assert_eq!(server.get_remote_static(), client_public);
+        let (client_res, server_res) = perform_handshake(&client, &server, server_public);
```

Comment:
I know this was already here, but `server_public_key` might be more clear to the reader.